common/src/main/java/com/microsoft/hydralab/performance/PerformanceTestManagementService.java

```diff
@@ -226,22 +226,22 @@ public void testFailure(String description) {
         inspectWithLifeCycle(InspectionStrategy.WhenType.TEST_FAILURE, description);
     }
 
-    public void testTearDown(TestRunDevice testRunDevice, TestTask testTask, TestRun testRun, Logger log) {
+    public void testTearDown(TestRunDevice testRunDevice, TestTask testTask, TestRun testRun) {
         List<ScheduledFuture<?>> timerList = inspectPerformanceTimerMap.get(testRun.getId());
         if (timerList != null) {
             for (ScheduledFuture<?> timer : timerList) {
                 timer.cancel(true);
             }
         }
         List<PerformanceTestResult> resultList = parseForTestRun(testRun);
-        savePerformanceTestResults(resultList, testRun, testTask, log);
+        savePerformanceTestResults(resultList, testRun, testTask, testRun.getLogger());
 
         inspectPerformanceTimerMap.remove(testRun.getId());
         testLifeCycleStrategyMap.remove(testRun.getId());
         testRunPerfResultMap.remove(testRun.getId());
 
         //TODO Android battery: adb shell dumpsys battery reset using Device info
-        LoggerFactory.getLogger(getClass()).info("Performance inspection finished");
+        testRun.getLogger().info("Performance inspection finished");
```

Comment:
Make sure testRun.getLogger() is non-null

Reply:
Fixed